source/simulator/assembly.cc

```diff
@@ -1419,7 +1419,8 @@ namespace aspect
                    * on the direction of the flow. Thus the flow-dependent terms below are
                    * only calculated if the edge is an inflow edge.
                    */
-                  const bool inflow = ((current_u * scratch.face_finite_element_values->normal_vector(q)) < 0.);
+                  const bool inflow = (advection_field.is_temperature()
+                                       || ((current_u * scratch.face_finite_element_values->normal_vector(q)) < 0.));
```

Comment:
@bangerth , can you explain more about your question?
"one half of the jump is simply the prescribed boundary term?"

I thought Sam's implementation for Dirichlet boundary seems right which are 

(u\cdot\grad T, \phi)_e  = - (\grad T, \phi u)_e +(u\cdot n T, \phi)_b  =  (u\cdot\grad T, \phi)_e -  (u\cdot n  T ^*, \phi)_b +(u\cdot n T, \phi)_b

on the boundary b, in the second term T^* uses the Dirichlet value and put into the RHS vector contribution, but  in the third term T put into the local matrix contribution